js/Inventory.html

```diff
@@ -0,0 +1,99 @@
+<script>
+/* global Item */
+/* exported Inventory */
+/**
+ * Inventory wraps an array with specialized functions
+ */
+function Inventory(array) {
+  let items = [];
+  this.setItems = function(array) {
+    if (! Array.isArray(array)) {
+      throw new Error(`${array} (${typeof array}) is not an array`);
+    }
+    items = array;
+    if (! (items[0] instanceof Item)) {
+      items.forEach(function (obj, index, array) {
+        array[index] = new Item(obj);
+      });
+    }
+  };
+  if (array) {
+    this.setItems(array);
+  }
+
+  // getLength exposes the array length
+  this.getLength = function() { return items.length; };
+
+  // this section exposes array functions
+  this.every = function(func) {
+    items.every(func);
+  };
+  this.forEach = function(func) {
+    items.forEach(func);
+  };
+  /**
+   * @returns Item[] - NOT Inventory!
```

Comment:
Let's confirm whether we want the returned object to be `Item[]` or `new Inventory`